### PR TITLE
Add all missing PHPDoc tags

### PR DESCRIPTION
**Pull request by thiemowmde**
I find PHPDoc tags for class properties extremely helpful, if not critical, because it saves one from scanning the whole file for the intended type.

I'm _not_ adding tags in cases where it's really, really obvious from the method signature.

### CODE REVIEW

File: src/Diff/PropertyDiffer.php [hunk 1]
```diff
@@ -72,10 +73,21 @@ public function diffProperties( Property $from, Property $to ) {
...
 	private function diffPropertyArrays( array $from, array $to ) {
 		return $this->recursiveMapDiffer->doDiff( $from, $to );
 	}
 
+	/**
+	 * @param Property $property
+	 *
+	 * @return array[]
+	 */
 	private function toDiffArray( Property $property ) {
```
Comment (Benestar) [inline src/Diff/PropertyDiffer.php:91]: In this case I think the signature is really really obvious ;-)

Comment (thiemowmde) [inline src/Diff/PropertyDiffer.php:91]: I disagree. How do you know "diff array" means `array[]`?

**Comment by JeroenDeDauw** [comment]
+1

File: src/Diff/Internal/FingerprintPatcher.php [hunk 2]
```diff
@@ -52,6 +52,11 @@ public function patchFingerprint( Fingerprint $fingerprint, EntityDiff $patch )
 		$this->patchAliases( $fingerprint, $patch->getAliasesDiff() );
 	}
 
+	/**
+	 * @param string[] $termArray
+	 *
+	 * @return TermList
+	 */
 	private function newTermListFromArray( $termArray ) {
```
Comment (JonasKress) [inline src/Diff/Internal/FingerprintPatcher.php]: Add a type hint?

Comment (JonasKress) [inline src/Diff/Internal/FingerprintPatcher.php]: +1

File: src/Lookup/EntityLookupException.php [hunk 3]
```diff
@@ -10,11 +12,19 @@
...
+	/**
+	 * @var EntityId
+	 */
 	private $entityId;
 
-	public function __construct( EntityId $entityId, $message = null, \Exception $previous = null ) {
+	/**
+	 * @param EntityId $entityId
+	 * @param string|null $message
+	 * @param Exception $previous
```
Comment (mariushoch) [inline src/Lookup/EntityLookupException.php]: Exception|null

File: src/Lookup/EntityRedirectLookupException.php [hunk 4]
```diff
@@ -10,11 +12,19 @@
...
+	/**
+	 * @var EntityId
+	 */
 	private $entityId;
 
-	public function __construct( EntityId $entityId, $message = null, \Exception $previous = null ) {
+	/**
+	 * @param EntityId $entityId
+	 * @param string|null $message
+	 * @param Exception $previous
```
Comment (mariushoch) [inline src/Lookup/EntityRedirectLookupException.php]: Exception|null

File: src/Lookup/ItemLookupException.php [hunk 5]
```diff
@@ -12,7 +13,12 @@
  */
 class ItemLookupException extends EntityLookupException {
 
-	public function __construct( ItemId $itemId, $message = null, \Exception $previous = null ) {
+	/**
+	 * @param ItemId $itemId
+	 * @param string|null $message
+	 * @param Exception $previous
```
Comment (mariushoch) [inline src/Lookup/ItemLookupException.php]: Exception|null

File: src/Lookup/LabelDescriptionLookupException.php [hunk 6]
```diff
@@ -10,11 +12,19 @@
...
+	/**
+	 * @var EntityId
+	 */
 	private $entityId;
 
-	public function __construct( EntityId $entityId, $message = null, \Exception $previous = null ) {
+	/**
+	 * @param EntityId $entityId
+	 * @param string|null $message
+	 * @param Exception $previous
```
Comment (mariushoch) [inline src/Lookup/LabelDescriptionLookupException.php]: Exception|null

File: src/Lookup/PropertyDataTypeLookupException.php [hunk 7]
```diff
@@ -10,11 +12,19 @@
...
+	/**
+	 * @var PropertyId
+	 */
 	private $propertyId;
 
-	public function __construct( PropertyId $propertyId, $message = null, \Exception $previous = null ) {
+	/**
+	 * @param PropertyId $propertyId
+	 * @param string|null $message
+	 * @param Exception $previous
```
Comment (mariushoch) [inline src/Lookup/PropertyDataTypeLookupException.php]: Exception|null

File: src/Lookup/PropertyLookupException.php [hunk 8]
```diff
@@ -12,7 +13,12 @@
  */
 class PropertyLookupException extends EntityLookupException {
 
-	public function __construct( PropertyId $propertyId, $message = null, \Exception $previous = null ) {
+	/**
+	 * @param PropertyId $propertyId
+	 * @param string|null $message
+	 * @param Exception $previous
```
Comment (mariushoch) [inline src/Lookup/PropertyLookupException.php]: Exception|null

File: src/Lookup/TermLookupException.php [hunk 9]
```diff
@@ -10,15 +12,24 @@
...
+	/**
+	 * @var EntityId
+	 */
 	private $entityId;
 
+	/**
+	 * @param EntityId $entityId
+	 * @param string[] $languageCodes
+	 * @param string|null $message
+	 * @param Exception $previous
```
Comment (mariushoch) [inline src/Lookup/TermLookupException.php]: Exception|null

File: src/Diff/Internal/FingerprintPatcher.php (hunk 2 above)
Comment (JeroenDeDauw) [inline src/Diff/Internal/FingerprintPatcher.php]: #wat. I forgot to add a type hint o_O

355zu7

**Comment by addshore** [comment]
+1

File: src/ByPropertyIdGrouper.php [hunk 10]
```diff
@@ -47,7 +47,10 @@ private function assertArePropertyIdProviders( $propertyIdProviders ) {
 		}
 	}
 
-	private function indexPropertyIdProviders( $propertyIdProviders ) {
+	/**
+	 * @param PropertyIdProvider[] $propertyIdProviders
+	 */
+	private function indexPropertyIdProviders( array $propertyIdProviders ) {
```
Comment (adrianheine) [inline src/ByPropertyIdGrouper.php:53]: #82

**Pull request merged**